Assets/Scripts/LifeSystem/LifeSystem.cs

```diff
@@ -0,0 +1,44 @@
+using UnityEngine;
+using UnityEngine.Events;
+
+public class LifeSystem : MonoBehaviour, IScoreSystem
+{
+    public int _score { get; set; } = 3;
+    public UnityEvent<int> _onValueChanged { get; set; }
+    public static LifeSystem Instance;
+    private PlayerSpawner _playerSpawner;
```

Comment:
```suggestion
    [SerializeField] private int _score { get; set; };
    [SerializeField] private UnityEvent<int> _onValueChanged { get; set; }
```
Alles kan private en dan in unity zelf de value 3 zetten. de unityEvent kan ook private omdat we het nooit direct aanroepen. check alles systems of de implemenaties van deze values private zijn. In de interface moet je namenlijk geen member access level maar in de derived classes wel

```suggestion
    [SerializeField] private UnityEvent _onDeath { get; set; }
    
    Public void CheckLifes()
    { 
         if (_score == 0)
         {
             _onDeath?.invoke();
         }
    }
```
Misschien een method zoals dit. het idee is dat we deze CheckLifes method assign aan de _OnValueChanged event. dus elke keer als de value changed checken we de score. als de score 0 is dan invoken we de event van _onDeath. dan kunnen ook later met UI een lose screen fixen dat de _onDeath event een lose screen aanroept